docs/modules/deployment/pages/upgrade/basic.adoc

```diff
@@ -0,0 +1,134 @@
+
+. Stop OpenNMS: `systemctl stop opennms.service`
++
+Expected result: `systemctl status opennms.service` reports opennms is stopped.
++
+NOTE: Your OpenNMS downtime starts at this point.
+
+. Stop earlier version of PostgreSQL: `systemctl stop postgresql`
++
+Expected result: `systemctl status postgresql.service` reports postgresql is stopped.
+
+. Disable earlier version PostgreSQL: `systemctl disable postgresql`
+. Stop PostgreSQL 12: `systemctl postgresql-12 stop`
++
+Expected result: `systemctl status postgresql-12.service` reports postgresql-12 is stopped.
+
+. Become the `postgres` user: `sudo su – postgres`
++
+Expected result: `whoami` returns `postgres`
+
+. Run the PostgreSQL upgrade in "check" mode: `/var/lib/pgsql/data -D /var/lib/pgsql/12/data -r -v -c`
++
+Expected result: output ends with “Clusters are compatible”.
++
+Verify these paths align with reality.
+
+. Run the PostgreSQL upgrade without "check": `/usr/pgsql-12/bin/pg_upgrade -b /usr/bin -B /usr/pgsql-12/bin -d /var/lib/pgsql/data -D /var/lib/pgsql/12/data -r -v`
++
+Expected outcome: output ends with “Upgrade Complete”.
++
+NOTE: Verify these paths align with reality.
+PostgreSQL's `pg_upgrade` utility copies the existing database into the new database and does not alter the existing database's data files.
+This leaves the PostgreSQL 9.4 database in its original state and location which can be used for rollback.
+
+. Replace `/var/lib/pgsql/12/data/postgresql.conf` with the updated copy.
+. Copy `pg_hba.conf` from the postgresql 9.4 data directory to the postgresql-12 data directory: `cp /var/lib/pgsql/data/pg_hba.conf /var/lib/pgsql/12/data/pg_hba.conf`
++
+NOTE: Verify these paths align with reality.
+. Exit PostgreSQL shell: `exit`
++
+Expected outcome: `whoami` returns `root`
+
+== Update and verify the {page-component-title} repository
+
+Your {page-component-title} repository is defined in a file in `/etc/yum.repos.d/`.
+It may be named {page-component-title}.repo  but is not guaranteed to be.
+
+. Use the text editor of your choice to edit your {page-component-title} repository config to change the `baseurl` of the repository from the old release family to the new release family.
+. Verify that the attribute `enabled=1` is defined, or enabled, or absent entirely
+. Purge any cached yum data: `yum clean all`
++
+Expected outcome: `yum repolist` does not show an error pertaining to the {page-component-title} repo; `yum info {page-component-title}-core` shows an available current package version.
+
+. Install the PostgreSQL 12 version of iplike: `yum -y install iplike-pgsql12`
++
+Expected outcome: `rpm -q iplike-pgsql12` reports an _iplike-pgsql12_ package is installed.
+
+. Start postgresql-12: `systemctl start postgresql-12.service`
++
+Expected outcome: `systemctl status postgresql-12.service` reports postgresql-12 has started.
+
+. Make a backup copy of your OpenNMS config: `mv /opt/opennms/etc /opt/opennms/etc.backup`
```

Comment:
`rsync -Ppav /opt/opennms/etc /tmp/etc.orig`
`rsync -Ppav /opt/opennms/jetty-webapps/opennms/WEB-INF /tmp/opennms-web-inf` 

This leaves the original `etc` in place and makes an identical copy in `/tmp`.  If they don't have a pre-upgraded config from Support, they get to handle reconciliation themselves so they'll need the config in place.